service_configuration_lib/spark_config.py

```diff
@@ -0,0 +1,106 @@
+import logging
+from typing import Any
+from typing import List
+from typing import Mapping
+from typing import MutableMapping
+from typing import Optional
+
+DEFAULT_SPARK_MESOS_SECRET_FILE = '/nail/etc/paasta_spark_secret'
+NON_CONFIGURABLE_SPARK_OPTS = {
+    'spark.master',
+    'spark.ui.port',
+    'spark.mesos.principal',
+    'spark.mesos.secret',
+    'spark.mesos.executor.docker.image',
+    'spark.mesos.executor.docker.parameters',
+    'spark.executorEnv.PAASTA_SERVICE',
+    'spark.executorEnv.PAASTA_INSTANCE',
+    'spark.executorEnv.PAASTA_CLUSTER',
+    'spark.hadoop.fs.s3a.access.key',
+    'spark.hadoop.fs.s3a.secret.key',
+    'fs.s3a.access.key',
+}
```

Comment:
how did you come up with this list?  I am noticing that it doesn't fully match what we have in spark-run (which doesn't have `fs.s3a.access.key`) or spark_tools (which includes `fs.s3a.access.key` and `spark.app.name` (although I'm not sure why)).

Reply:
Ummm I just kinda guessed.  I've taken `fs.s3a.access.key` out for now, and it "seems" OK to allow users to set the `spark.app.name`?

I also just added `spark.executorEnv.SPARK_EXECUTOR_DIRS` to this list, which it seems like maybe we don't want to let people set?

Maybe @huadongliu can comment on this list.